Add support to load Tiled maps

danielbarnes175 opened the pull request:
## Summary
Adds support to load maps created with https://www.mapeditor.org/
Also adds a directory that contains the tiled project

## Technical Changes
- Add various mapping classes in `/Engine/Map/` that mirror the Tiled JSON structure, so that we can directly load into it
- Combine Program.cs into Game1.cs and renamed Main.cs to make it cleaner
- Add example loading to Main.cs
- Add Tiled map directory, outside of the main game directory so we don't export these files when publishing
- Add Newtonsoft.Json as a dependency to make loading JSON easier

## Screenshots
![image](https://github.com/maxdevos49/gate-adventures/assets/31489642/c0f08414-4f72-4fd8-bc59-5df7fd2160e9)

danielbarnes175 committed:
Add Tiled boilerplate

danielbarnes175 committed:
Add Newtonsoft.Json

danielbarnes175 committed:
Add classes to map Tiled JSON

danielbarnes175 committed:
Update main to load tile map into game

danielbarnes175 committed:
Format

danielbarnes175 committed:
Update packages lock file

danielbarnes175 committed:
Fix formatting

danielbarnes175 committed:
Update Tiled support for multiple tilesets and a TiledMapLoader

maxdevos49 commented on GateAdventures/Engine/Map/TiledMapLoader.cs:
```diff
@@ -0,0 +1,53 @@
+	public static TileMapRenderer LoadSingleMap(ContentManager content, GraphicsDevice graphics, string mapFilePath)
+	{
+		// Load JSON into TileMap object
+		string filePath = Path.Combine(content.RootDirectory, mapFilePath);
```
When I run `dotnet run --project GateAdventure` it fails to find the JSON file.

The resulting path seems to be: `<path-to-where-command-is-run>` + `<content.RootDirectory>` + `<mapFilePath>`.

If I run `dotnet run` inside the GateAdventures folder it works. I think it should work in both situations and not require being inside the Project folder. Desired path ending up being: `<ExecutableFolderPath>` + `<content.RootDirectory>` + `<mapFilePath>`.

maxdevos49 commented on GateAdventures/Game1.cs:
```diff
@@ -1,51 +0,0 @@
-﻿using Microsoft.Xna.Framework;
```
I agree I am not a fan of this class being named Game1 but I did like the Program.cs file. What if the `Game1` class was renamed `GateAdventures` and Program.cs was preserved?

maxdevos49 commented on GateAdventures/Engine/Map/Layers.cs:
```diff
@@ -0,0 +1,20 @@
+﻿using System;
```
Mostly a nit but I think we should drop `Engine` in this file path

maxdevos49 replied:
I retract this

danielbarnes175 replied:
Dang it!!

maxdevos49 replied:
Sorry D:

maxdevos49 reviewed (dismissed):
@danielbarnes175 can you add some brief instructions on how to use the tile map editor to the readme

danielbarnes175 committed:
Address review comments

danielbarnes175 commented:
@maxdevos49 Comments addressed,
- Added program.cs back in, renamed Main.cs to GateAdventures.cs
- Adjusted the path of the file to use `Path.Combine(AppContext.BaseDirectory, content.RootDirectory, mapFilePath);`
- Got rid of the `Engine` folder and added it back 🙊 
- Added a new `README.md` in the Tiled folder since it made more sense to me to have it separate from the main one. There were also some additional notes I needed to add about how to actually import the map

maxdevos49 committed:
Update CHANGELOG.md
